src/main/scala/kinesis/mock/api/GetShardIteratorRequest.scala

```diff
@@ -95,21 +95,34 @@ final case class GetShardIteratorRequest(
                             InvalidArgumentException(
                               s"Timestamp cannot be in the future"
                             ).invalidNel
-                          else
+                          else {
+                            val sequenceNumber =
+                              data
+                                .find(
+                                  _.approximateArrivalTimestamp.toEpochMilli > ts.toEpochMilli
```

Comment:
GetShardIterator was bugged if you passed in a timestamp (e.g. AT_TIMESTAMP). Some fixes for this are here.